### PR TITLE
Added basic filter for CLI logging.

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/18560|

## Description

Adds Filter to loggers that are executed in the CLI.

### CODE REVIEW

File: framework/wazuh/core/tests/test_wlogging.py [hunk 1]
```diff
@@ -219,3 +219,24 @@ def __init__(self, log_type):
...
     cf = wlogging.CustomFilter('testA')
     assert not cf.filter(MockedRecord('testB'))
+
+
+@pytest.mark.parametrize('value, expected',
+                         [
+                             ('Example log', True),
+                             ('Wazuh Internal Error', False),
+                             ('WazuhInternalError', False)
+                         ])
```
Comment (GGP1) [inline framework/wazuh/core/tests/test_wlogging.py:229]:
```suggestion
@pytest.mark.parametrize('value, expected', [
    ('Example log', True),
    ('Wazuh Internal Error', False),
    ('WazuhInternalError', False),
    ('WazuhError', True),
    ('InternalError', True)
])
```

File: framework/wazuh/core/wlogging.py [hunk 2]
```diff
@@ -237,3 +237,29 @@ def __getattr__(self, item: str) -> object:
...
+        Parameters
+        ----------
+        record : logging.LogRecord
+            Contains all the information to the event being logged.
+
+        Returns
+        -------
+        bool
+            Boolean used to determine if the log entry should be logged.
+        """
```
Comment (GGP1) [inline framework/wazuh/core/wlogging.py:261]:
```suggestion
    def filter(self, record: logging.LogRecord) -> bool:
        """Filter the log entry depending on its message contents.

        Parameters
        ----------
        record : logging.LogRecord
            Contains the information of the event being logged.

        Returns
        -------
        bool
            Whether the log entry should be logged or not.
        """
```

**Review by GGP1 — CHANGES_REQUESTED** [review]

File: framework/wazuh/core/tests/test_wlogging.py (hunk 1 above)
Comment (RamosFe) [reply framework/wazuh/core/tests/test_wlogging.py]: Fixed in 08f6d1a3a8a5db68c5dddd11bc1b88a047614105.

**Review by GGP1 — APPROVED** [review]
LGTM

**Review by Selutario — APPROVED** [review]
LGTM. According to the example shown [here](https://github.com/wazuh/wazuh/issues/18560#issuecomment-2145606882), less accurate information (`Agent 002 status: Send upgrade command error`) would be shown now if compared to the previous message (`Error 500: The WPK package used does not fit with the agent's architecture`) though.

**Pull request merged**